crates/primitives/src/bytes/mod.rs

```diff
@@ -21,21 +21,25 @@ pub struct Bytes(pub bytes::Bytes);
 
 impl fmt::Debug for Bytes {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_str("Bytes(")?;
-        f.write_str(&self.hex_encode())?;
-        f.write_str(")")
+        fmt::LowerHex::fmt(self, f)
     }
 }
 
 impl fmt::Display for Bytes {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_str(&self.hex_encode())
+        fmt::LowerHex::fmt(self, f)
     }
 }
 
 impl fmt::LowerHex for Bytes {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_str(&self.hex_encode())
+        f.pad(&hex::encode_prefixed(self.as_ref()))
```

Comment:
we should match std behavior here by adding `0x` only  when `fmt.alternate()` is true

Reply:
(as a drive-by while we're touching this code)

Reply:
I'll open a new PR for this